app/src/main/java/com/github/se/gomeet/viewmodel/UserViewModel.kt

```diff
@@ -196,6 +196,37 @@ class UserViewModel(val currentUID: String? = null) : ViewModel() {
     }
   }
 
+  /**
+   * User deletes an event and removes it from their list of myEvents. It is used when a user
+   * deletes an event.
+   *
+   * @param eventId The id of the event to delete.
+   * @param userId The id of the user deleting the event.
+   */
+  suspend fun userDeletesEvent(eventId: String, userId: String = currentUID!!) {
```

Comment:
This is something that we have done from the start but is a mistake. Suspend functions shouldn't exist in ViewModels, because they get called in Composable functions. I'm not sure whether it really changes anything performance-wise, but I discovered that if you want to call suspend functions in a class extending the basic ViewModel class, you can use `viewModelScope.launch { /* async stuff going on */ }`. This would be a lot cleaner and more correct convention-wise. From my understanding, it is also the case that a suspend function launched in a coroutine will get cancelled as soon as that coroutine "dies" or is refreshed. I.e. when you call a suspend function in a coroutineScope inside a Composable function, then as soon as that Composable gets refreshed (or if the user navigates to another screen), the suspend function launched in the coroutineScope is cancelled if it hasn't completed yet.

Reply:
Well, it works for now we will do the refactoring later